File: 240722/부분 수열의 합이 k/the-sum-of-the-subsequences-is-k.js
const fs = require('fs');
const input = fs.readFileSync(0).toString().trim().split('\n');

const [n, k] = input[0].trim().split(" ").map(Number);
const nums = input[1].trim().split(" ").map(Number);

let prefixSums = Array(n).fill(0);
prefixSums[0] = nums[0];

for (let i = 1; i < n; i++) {
    prefixSums[i] = prefixSums[i - 1] + nums[i];
}

let count = 0;

for (let i = 0; i < n; i++) {
    for (let j = i; j < n; j++) {
        if (i === j && nums[i] === k) count++;
        else if (i === 0 && prefixSums[j] === k) count++;
        else if (prefixSums[j] - prefixSums[i - 1] === k) count++; 
    }
}

console.log(count);

File: 240722/부분 수열의 합이 k/the-sum-of-the-subsequences-is-k.kt
import java.io.*

fun main() {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val nums =
        br
            .readLine()
            .trim()
            .split(" ")
            .map { it.toInt() }
    val arr =
        br
            .readLine()
            .trim()
            .split(" ")
            .map { it.toInt() }
    var count = 0
    var now = 0
    var start = 0
    for (j in 0 until nums[0]) {
        now += arr[j]
        while (now > nums[1]) {
            now -= arr[start]
            start++
        }
        if (now == nums[1]) {
            count++
        }
    }
    print(count)
}